Libplanet/Store/Trie/MerkleTrie.cs

```diff
@@ -0,0 +1,333 @@
+            }
+        }
+
+        private bool TryGet(
+            INode node,
+            ImmutableArray<byte> prefix,
+            ImmutableArray<byte> path,
+            out IValue value)
+        {
+            switch (node)
+            {
+                case null:
+                    value = null;
+                    return false;
+
+                case ValueNode valueNode:
+                    value = valueNode.Value;
+                    return true;
+
+                case ShortNode shortNode:
+                    if (path.Length < shortNode.Key.Length
+                        || !path.Take(shortNode.Key.Length).SequenceEqual(shortNode.Key))
+                    {
+                        value = null;
+                        return false;
+                    }
+
+                    return TryGet(
+                        shortNode.Value,
+                        prefix.AddRange(path.Take(shortNode.Key.Length)),
+                        path.Skip(shortNode.Key.Length).ToImmutableArray(),
+                        out value);
+
+                case FullNode fullNode:
+                    INode childNode = fullNode.Children[path[0]];
+                    return TryGet(
+                        childNode,
+                        prefix.Add(path[0]).ToImmutableArray(),
+                        path.Skip(1).ToImmutableArray(),
+                        out value);
+
+                case HashNode hashNode:
+                    try
+                    {
+                        INode resolvedNode = GetNode(hashNode.HashDigest);
+                        return TryGet(resolvedNode, prefix, path, out value);
+                    }
+                    catch (KeyNotFoundException)
+                    {
+                        value = null;
+                        return false;
+                    }
+
+                default:
+                    throw new InvalidTrieNodeException(
+                        $"Invalid node: raw: {node.ToBencodex().Inspection}");
+            }
+        }
+
+        /// <summary>
+        /// Gets the node corresponding to <paramref name="nodeHash"/> from storage,
+        /// (i.e., <see cref="KeyValueStore"/>).
+        /// </summary>
+        /// <param name="nodeHash">The hash of node to get.</param>
+        /// <returns>The node corresponding to <paramref name="nodeHash"/>.</returns>
+        private INode GetNode(HashDigest<SHA256> nodeHash)
+        {
+            return NodeDecoder.Decode(
+                _codec.Decode(KeyValueStore.Get(nodeHash.ToByteArray())));
+        }
+
+        private byte[] ToKey(byte[] key)
+        {
+            if (_secure)
+            {
+                key = Hashcash.Hash(key).ToByteArray();
+            }
+
+            var res = new byte[key.Length * 2];
+            for (var i = 0; i < key.Length; ++i)
+            {
+                res[i * 2] = (byte)(key[i] >> 4);
+                res[i * 2 + 1] = (byte)(key[i] & 0b00001111);
```

Comment:
How about making `0b00001111` to a constant variable?

Reply:
I understood your comment like below:

```csharp
            const int lowerBytesMask = 0b00001111;
            for (var i = 0; i < key.Length; ++i)
            {
                res[i * 2] = (byte)(key[i] >> 4);
                res[i * 2 + 1] = (byte)(key[i] & lowerBytesMask);
            }
```

Are the codes you intended? Then IMHO I'm not sure what the point makes it better. Is it for readability?

Reply:
Yeah, I thought that's better for readability.

Reply:
I applied it in 199a7f9. Could you check it again?